src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -149,7 +161,55 @@ public void updateFilteredPersonList(Predicate<Person> predicate) {
         requireNonNull(predicate);
         filteredPersons.setPredicate(predicate);
     }
+    //=========== LeavesBook ================================================================================
+    public ReadOnlyLeavesBook getLeavesBook() {
+        return leavesBook;
+    }
+    @Override
+    public void setLeavesBook(ReadOnlyLeavesBook leavesBook) {
+        this.leavesBook.resetData(leavesBook);
+    }
+
+    //=========== Filtered Leave List Accessors =============================================================
+
+    @Override
+    public boolean hasLeave(Leave leave) {
+        requireNonNull(leave);
+        return leavesBook.hasLeave(leave);
+    }
+
+    @Override
+    public void deleteLeave(Leave target) {
+        leavesBook.removeLeave(target);
+    }
 
+    @Override
+    public void addLeave(Leave leave) {
+        leavesBook.addLeave(leave);
+        updateFilteredLeaveList(PREDICATE_SHOW_ALL_LEAVES);
```

Comment:
Is it our desired behaviour to reset all filters for the leaves list?

Reply:
for now